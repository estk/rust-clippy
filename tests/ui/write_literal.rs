// #![feature(plugin)]
// #![plugin(clippy)]
#![allow(unused_must_use)]
// #![warn(write_literal)]
use std::io::Write;

fn main() {
    let mut v = Vec::new();
    // // these should be fine
    write!(&mut v, "Hello");
    writeln!(&mut v, "Hello");
    let world = "world";
    writeln!(&mut v, "Hello {}", world);
    writeln!(&mut v, "3 in hex is {:X}", 3);

    // these should throw warnings
    write!(&mut v, "Hello {}", "world");
    writeln!(&mut v, "Hello {} {}", world, "world");
    writeln!(&mut v, "Hello {}", "world");
    writeln!(&mut v, "10 / 4 is {}", 2.5);
    writeln!(&mut v, "2 + 1 = {}", 3);
    // writeln!(&mut v, "2 + 1 = {:.4}", 3);
    // writeln!(&mut v, "2 + 1 = {:5.4}", 3);
    // writeln!(&mut v, "Debug test {:?}", "hello, world");

    // positional args don't change the fact
    // that we're using a literal -- this should
    // throw a warning
    // writeln!("{0} {1}", "hello", "world");
    // writeln!("{1} {0}", "hello", "world");

    // named args shouldn't change anything either
    // writeln!("{foo} {bar}", foo = "hello", bar = "world");
    // writeln!("{bar} {foo}", foo = "hello", bar = "world");
}
