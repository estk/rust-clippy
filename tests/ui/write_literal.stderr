error: writing a literal with an empty format string
  --> $DIR/write_literal.rs:17:32
   |
17 |     write!(&mut v, "Hello {}", "world");
   |                                ^^^^^^^
   |
   = note: `-D write-literal` implied by `-D warnings`

error: writing a literal with an empty format string
  --> $DIR/write_literal.rs:18:44
   |
18 |     writeln!(&mut v, "Hello {} {}", world, "world");
   |                                            ^^^^^^^

error: writing a literal with an empty format string
  --> $DIR/write_literal.rs:19:34
   |
19 |     writeln!(&mut v, "Hello {}", "world");
   |                                  ^^^^^^^

error: writing a literal with an empty format string
  --> $DIR/write_literal.rs:20:38
   |
20 |     writeln!(&mut v, "10 / 4 is {}", 2.5);
   |                                      ^^^

error: writing a literal with an empty format string
  --> $DIR/write_literal.rs:21:36
   |
21 |     writeln!(&mut v, "2 + 1 = {}", 3);
   |                                    ^

error: aborting due to 5 previous errors

